{{ cookiecutter.project_slug }}/tests/strategies/test_download.py

```diff
@@ -0,0 +1,50 @@
+"""Tests the download strategy for 'file://'."""
+# pylint: disable=invalid-name
+from typing import TYPE_CHECKING
+
+import pytest
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from oteapi.models import SessionUpdate
+
+    from {{ cookiecutter.package_name }}.strategies.download import SessionUpdateFile
+
+
+@pytest.mark.parametrize(
+    "filename,mediaType",
+    [("sample1.db", "application/vnd.sqlite3"), ("sample2.json", "application/json")],
+    ids=["binary", "text"],
+)
+def test_download_file(filename: str, mediaType: str, static_files: "Path") -> None:
+    """Test `file` download strategy on binary and text files.
+
+    Test files are taken from filesamples.com.
+    """
+    from oteapi.datacache import DataCache
+
+    from {{ cookiecutter.package_name }}.strategies.download import FileStrategy
+
+    sample_file = static_files / filename
+    assert sample_file.exists(), f"Test file not found at {sample_file} !"
+
+    config = {
+        "downloadUrl": sample_file.as_uri(),
+        "mediaType": mediaType,
+    }
+    download_initialize: "SessionUpdate" = FileStrategy(config).initialize()
+    download_get: "SessionUpdateFile" = FileStrategy(config).get(download_initialize)
+
+    content: bytes = DataCache().get(download_get.key)
+
+    assert {**download_initialize} == {}
```

Comment:
I am unfamiliar with this syntax. How is this different from assert download_initialize == {}?

Reply:
The main issue/test here is that `download_initialize` is not actually a dictionary, but rather a `SessionUpdate` instance. This means by definition `download_initialize` will not be equal to `{}`.
However, we have implemented the ability for it to unpack similarly as a dictionary when adding two asterisks prior to it `**download_initialize`, and we also want to test that it can do that.
But since a bare statement `**download_initialize` is not allowed (it fails when one does this) it needs to be packed into a dictionary, hence the surrounding `{}`.

Reply:
Thanks for the explanation. Should it be added as a comment in the code?

Reply:
I see. Thanks, Casper! I think a short comment on this in the code would be helpful.